src/main/java/com/snowflake/kafka/connector/internal/streaming/telemetry/SnowflakeTelemetryChannelStatus.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2023 Snowflake Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.snowflake.kafka.connector.internal.streaming.telemetry;
+
+import static com.snowflake.kafka.connector.internal.metrics.MetricsUtil.constructMetricName;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.MetricRegistry;
+import com.snowflake.kafka.connector.internal.KCLogger;
+import com.snowflake.kafka.connector.internal.metrics.MetricsJmxReporter;
+import com.snowflake.kafka.connector.internal.metrics.MetricsUtil;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class SnowflakeTelemetryChannelStatus {
+  private static final KCLogger LOGGER =
+      new KCLogger(SnowflakeTelemetryChannelStatus.class.toString());
+
+  public static final long NUM_METRICS = 3; // update when new metrics are added
+  private static final long INITIAL_OFFSET = -1;
+
+  // channel properties
+  private final String channelName;
+  private final MetricsJmxReporter metricsJmxReporter;
+
+  private final AtomicLong startTime;
+
+  // offsets
+  private AtomicLong offsetPersistedInSnowflake;
+  private AtomicLong processedOffset;
+  private long latestConsumerOffset;
```

Comment:
do we need atomic long?

Reply:
SnowflakeTelemetryPipeStatus uses AtomicLongs, so I figured this should follow precedent

Reply:
removed, register the guages with TopicPartitionChannel offsets directly instead